src/__tests__/groupMembership.test.ts

```diff
@@ -0,0 +1,139 @@
+import Account from '../account'
+// @ts-ignore no type defs exist for js-sdk
+import Tozny from '@toznysecure/sdk/node'
+import { v4 as uuidv4 } from 'uuid'
+import { cleanupRealms } from './utils'
+
+// default of 5s is sometimes, but not always, enough time. give the tests 10s
+let testTimeout = Number(process.env.TEST_TIMEOUT)
+jest.setTimeout(testTimeout)
+
+const accountFactory = new Account(Tozny, process.env.API_URL)
+let client: any = null
+let realmName: string
+let identity: any = null
+let realmConfig: any = null
+let realm: any = null
+let username: any = null
+let apiURL = process.env.API_URL
+
+beforeAll(async () => {
+  // Create an account to re-use across test cases
+  const seed = uuidv4()
+  const name = `Test Account ${seed}`
+  const email = `test+${seed}@tozny.com`
+  const password = uuidv4()
+  const accountClient: any = await accountFactory.register(
+    name,
+    email,
+    password
+  )
+  const sovereignName = 'JELP'
+  realmName = `TestRealm${seed.split('-')[0]}`.toLowerCase()
+
+  const tokenName = 'example token'
+  const permissions = {
+    enabled: true,
+    allowed_types: ['general', 'identity'],
+  }
+  client = accountClient.client
+  const token = await client.newRegistrationToken(tokenName, permissions)
+  const stringToken = String(token.token)
+  const createdRealm = await client.createRealm(realmName, sovereignName)
+  username = `it-user-${uuidv4()}`
+  realmConfig = {
+    realmName: realmName,
+    appName: 'account',
+    brokerTargetUrl: 'http://integrationtest.local.tozny.com',
+    apiURL,
+  }
+  realm = new Tozny.identity.Realm(
+    realmConfig.realmName,
+    realmConfig.appName,
+    realmConfig.brokerTargetUrl,
+    apiURL
+  )
+  await realm.register(
+    username,
+    password,
+    stringToken,
+    `${username}@example.com`
+  )
+  identity = await realm.login(username, password)
+})
+
+afterAll(async () => {
+  await cleanupRealms(client)
+})
+
+describe('Identity Group Membership', () => {
+  it('adds, lists, removes identity group membership', async () => {
+    let identity_id = identity.storage.config.clientId
+
+    // This identity should have no groups
+    const responseEmptyGroups = await client.groupMembership(
+      realmName,
+      identity_id
+    )
+    expect(responseEmptyGroups).toBeInstanceOf(Array)
+    expect(responseEmptyGroups).toHaveLength(0)
+
+    // Create a group
+    const group1 = await client.createRealmGroup(realmName, {
+      name: 'ToznyEngineers',
+    })
+
+    // Check for group created
+    expect(group1.id).toBeTruthy()
+    expect(group1.name).toBe('ToznyEngineers')
+
+    // Add identity to the group
+    await client.updateGroupMembership(realmName, identity_id, {
+      groups: [group1.id],
```

Comment:
the type signature for `groups` says it takes objects that look like the `Group` class but you're passing in strings. does it support both?

if so, we can indicate that in the types with `type GroupsInput = Group | string`

i've noticed that the api actually only requires a subset of the properties for its inputs in request bodies. when i notice that's the case, i have been separating the input type from the actual class types. i have an example in `groupRoleMappings` where the API actually only wants a subset of the actual `Role` objects.

what you have will work for our setup, but when we eventually expose ts typing to the world it will create problems because the functions will be requiring more properties than are actually needed, and in some cases more than can be provided